src/pykeen/training/accelerated_training_loop.py

```diff
@@ -0,0 +1,454 @@
+# -*- coding: utf-8 -*-
+
+"""Accelerated Training loop for KGE models using HF Accelerate"""
+
+import gc
+import logging
+import pathlib
+import time
+from abc import ABC
+from typing import Any, List, Mapping, Optional, Type, Union
+
+import torch
+from accelerate import Accelerator
+from torch.optim.optimizer import Optimizer
+from torch.utils.data import DataLoader
+from tqdm.autonotebook import tqdm, trange
+
+from .training_loop import NonFiniteLossError, SubBatchingNotSupportedError, TrainingLoop, _get_optimizer_kwargs
+from ..losses import Loss, has_mr_loss
+from ..models import Model, RGCN
+from ..stoppers import Stopper
+from ..trackers import ResultTracker
+from ..training.schlichtkrull_sampler import GraphSampler
+from ..triples import Instances
+from ..utils import format_relative_comparison, get_batchnorm_modules
+
+__all__ = [
+    'AcceleratedTrainingLoop',
+]
+
+logger = logging.getLogger(__name__)
+
+
+class AlternateAcceleratedTrainingLoop(TrainingLoop, ABC):
```

Comment:
This is an alternative to `AcceleratedTrainingLoop`, I guess? What is the downside of this implementation? It looks a lot cleaner and seems to avoid almost completely code duplication.

Reply:
There is still one thing I'm not sure how should be abstracted

Reply:
looks much cleaner indeed!

Reply:
There is also some accelerate-specific stuff with `tqdm` args to prevent spawning multiple bars though, but can be postponed

Reply:
I think I addressed this in 998b822c